Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -121,10 +121,31 @@ extension StripeCardReaderService: CardReaderService {
             Terminal.shared.simulatorConfiguration.simulatedCard = .init(type: .amex)
         }
 
-        let config = DiscoveryConfiguration(
-            discoveryMethod: discoveryMethod.toStripe(),
-            simulated: shouldUseSimulatedCardReader
-        )
+        let config: DiscoveryConfiguration
+        switch discoveryMethod {
+        case .bluetoothScan:
+            let blueToothConfig = BluetoothScanDiscoveryConfigurationBuilder()
+            do {
+                config = try blueToothConfig.setSimulated(shouldUseSimulatedCardReader).build()
+            } catch let error as UnderlyingError {
+                DDLogError("Failed to start BluetoothScanDiscovery. Error:\(String(describing: error.failureReason))")
+                throw error
+            } catch {
+                DDLogError("\(error)")
+                throw error
+            }
+        case .localMobile:
+            let localMobileConfig = LocalMobileDiscoveryConfigurationBuilder()
+            do {
+                config = try localMobileConfig.setSimulated(shouldUseSimulatedCardReader).build()
+            } catch let error as UnderlyingError {
+                DDLogError("Failed to start LocalMobileDiscovery. Error:\(String(describing: error.failureReason))")
+                throw error
+            } catch {
+                DDLogError("\(error)")
+                throw error
+            }
```

Comment:
Seems like perhaps this could use a common method which takes the appropriate ConfigurationBuilder as an argument?

Reply:
That's a good point. Since we do not have a common public interface that we can use as an argument, we have to create our own. Added to this, I found some problems when trying to extend the builders to conform to our own protocol (eg: their classes are not final, so some of the methods we need to use cannot return Self), which left me with creating wrappers around them and using those.

I created a different branch here: https://github.com/woocommerce/woocommerce-ios/commit/c3279fb0c0c22024f3d3a07819576309626bec39 to show the changes, since I'm unsure if we should leave it as it is for the moment, or implement those. I like the idea of the wrappers since abstracts the builder and removes duplication, but also seems to add a bit of overhead. Thoughts? If seems a good idea I can iterate further and submit a new PR with those.

Reply:
Sorry for the delay in the re-review. I wanted to address this comment/thread before I approved, but then ran out of time. I'll approve this PR now and cover the builder changes separately later on today.